spring-cloud-gcp-autoconfigure/src/main/java/com/google/cloud/spring/autoconfigure/sql/DatabaseType.java

```diff
@@ -36,26 +36,44 @@
 	 */
 	POSTGRESQL("org.postgresql.Driver", "jdbc:postgresql://google/%s?"
 			+ "socketFactory=com.google.cloud.sql.postgres.SocketFactory"
-			+ "&cloudSqlInstance=%s", "postgres");
+			+ "&cloudSqlInstance=%s", "postgres"),
+
+	/**
+	 * R2DBC MySQL constants.
+	 */
+	R2DBC_MYSQL(null,  "r2dbc:gcp:mysql://%s/%s", "root"),
```

Comment:
It's not what I had in mind, but maybe it works better? I was thinking of the enum just having 4 fields: `jdbcDriver`, `jdbcUrlTeample`, `r2dbcUrlTemplate`, `defaultUsername`. In my opinion it makes more sense. Does it not work well?
Also, notice that semantically `R2DBC_MYSQL` is not really a "database type". The fact that you need to pass a `null` for the driver class is also a smell that something is off in design.

Reply:
This came from my review -- having both `jdbcUrlTemplate` and `r2dbcUrlTemplate` is awkward because one of them is always set and the other is always null. They really represent the same concept -- the URL pattern, so I asked to collapse them. Funny, the "passing null in constructor" was also what tripped me up, but for the URL template field.

For the jdbcDriver field, there does not seem to be an applicable concept in R2DBC since the driver is inferred from the URL pattern. Not sure how you'd avoid setting it to null.

Reply:
Ah, I misunderstood -- I was thinking of enum having 4 constants. But if the enum only has 2 constants (JDBC and R2DBC), then having both, jdbcUrl and r2dbcUrl makes sense.

Reply:
Sorry, two types -- MySQL and Postgres; not JDBC and R2DBC.

Reply:
My bad, I was overthinking this. The presence of `jdbcDriver` for r2dbc confused me a little. However, have just `MySQL` and `Postgres` (as you mentioned) with appropriately named getters makes a lot of sense. Thank you!